### PR TITLE
added method to retrieve private port number for allowed ip

### PR DESCRIPTION
**Pull request by gspurthi**
Associated with the issue: 197

Changes included in the PR: Updated a method to retrieve private port number for allowed ip.

### CODE REVIEW

File: instances.go [hunk 1]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
...
+		return nil, err
+	}
+
+	_, err = c.do(ctx, req, &portResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(portResponse.Ports) == 0 {
+		return nil, fmt.Errorf("No port number available")
```
Comment (stephaniegalang-ibm) [inline instances.go:282]: Could be more specific
```suggestion
		return nil, fmt.Errorf("No port number available, check that the instance has an enabled allowedIP policy")
```

**Review by stephaniegalang-ibm — APPROVED** [review]
lgtm! Have a suggestion for the error msg

File: instances.go [hunk 2]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
 
 	return nil
 }
+
+type PortsMetadata struct {
+	CollectionType string `json:"collectionType"`
+	NumberOfPorts  int    `json:"collectionTotal"`
+}
+
+type Ports struct {
```
Comment (mrodden) [inline instances.go:259]:
```suggestion
type portResponse struct {
```

File: instances.go [hunk 3]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
...
+	Metadata PortsMetadata `json:"metadata"`
+	Ports    []PrivatePort `json:"resources"`
+}
+type PrivatePort struct {
+	PrivatePort int `json:"private_endpoint_port,omitempty"`
+}
+
+// GetAllowedIPPrivateNetworkPort retrieves the private endpoint port assigned to allowed ip policy.
+func (c *Client) GetAllowedIPPrivateNetworkPort(ctx context.Context) (*PrivatePort, error) {
+	var portResponse Ports
```
Comment (mrodden) [inline instances.go:269]:
```suggestion
	var resp portResponse
```

File: instances.go [hunk 4]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
 
 	return nil
 }
+
+type PortsMetadata struct {
```
Comment (mrodden) [inline instances.go:254]:
```suggestion
type portsMetadata struct {
```

File: instances.go [hunk 5]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
...
+	CollectionType string `json:"collectionType"`
+	NumberOfPorts  int    `json:"collectionTotal"`
+}
+
+type Ports struct {
+	Metadata PortsMetadata `json:"metadata"`
+	Ports    []PrivatePort `json:"resources"`
+}
+type PrivatePort struct {
```
Comment (mrodden) [inline instances.go:263]:
```suggestion
type privatePort struct {
```

File: instances.go [hunk 6]
```diff
@@ -249,3 +250,36 @@ func (c *Client) SetInstancePolicies(ctx context.Context, setDualAuth, dualAuthE
...
+
+	_, err = c.do(ctx, req, &portResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(portResponse.Ports) == 0 {
+		return nil, fmt.Errorf("No port number available. Please check the instance has an enabled allowedIP policy")
+	}
+	return &portResponse.Ports[0], nil
```
Comment (mrodden) [inline instances.go:284]:
```suggestion
	return &portResponse.Ports[0].PrivatePort, nil
```

File: kp_test.go [hunk 7]
```diff
@@ -1686,6 +1686,57 @@ func TestSetAllowedIPPolicyError(t *testing.T) {
 
 	assert.Error(t, err)
 	assert.Equal(t, "Please provide at least 1 IP subnet specified with CIDR notation", err.Error())
+
+}
+
+// TestGetPrivateEndpointPortNumber tests the method that retrieves the private endponint port number
```
Comment (mrodden) [inline kp_test.go:1692]:
```suggestion
// TestGetPrivateEndpointPortNumber tests the method that retrieves the private endpoint port number
```

**Review by mrodden — APPROVED** [review]

**Pull request merged**